src/OVAL/probes/probe/entcmp.c

```diff
@@ -36,8 +36,12 @@
 #include <string.h>
 #include <stdarg.h>
 #include <errno.h>
+#ifdef _WIN32
```

Comment:
By skimming through `winsock2.h`, I got the impression that it exposes different API. Therefore, I would shift this platform-dependency one level up - instead of having `#ifdef`s in the source file, have the naturally cross-platform part of `entcmp.c` stay there, separate the unix-only part to `entcmp-unix.c` and introduce a windows-only `entcmp-w32.c`. Then, conditionally insert the right `.c` file into the compilation by CMake.

Reply:
I don't prefer creating 2 almost identical files that will differ only in 1-2 lines. That will be harder to maintain, because if somebody will fix or change this code in future, he will have to change the same code at 2 places, and he will very likely forget to change one of them.

Reply:
I meant something else than creating almost files, but since this can be done any time later, I am fine with either way for now.

Reply:
@matejak What was your idea then?

Reply:
My idea was to keep the common stuff in one file and the platform-specific stuff in another two files. Therefore, this is not about having multiple almost identical files, but it is probably more about implementing a function with platform-independent interface, but platform-specific internals in those two files.

Reply:
@matejak OK, but is there something platform specific in the `entcmp.c`?